python/lsst/ts/Dome/task_scheduler.py

```diff
@@ -16,20 +15,9 @@ async def schedule_task_periodically(period, task):
     task: coroutine
         The function to be scheduled periodically.
     """
-    loop = asyncio.get_event_loop()
     log.info(f"run_status_loop = {run_status_loop}")
 
-    def g_tick():
-        t = loop.time()
-        count = 0
-        while run_status_loop:
-            count += 1
-            yield max(t + count * period - loop.time(), 0)
-
-    g = g_tick()
-
     while run_status_loop:
```

Comment:
I strongly suggest deleting this module and ditching the global. I think the code would be much clearer as a function.
```
async def status_loop(self, period):
    while self.run_status_loop:
        self.status()
        await asyncio.sleep(period)
```

In addition, please consider canceling the task when you want to close it instead of setting a global or instance variable. The advantage is that you eliminate the latency of waiting for the sleep to finish when you want the loop to end.

That is the pattern I describe here: https://ts-salobj.lsst.io/salobj_cscs.html#telemetry-loop-example

A counter-indicator is if cancel could interrupt something that should not be interrupted -- in this case that would be something in your `status` function. But the only thing that could be interrupted is a TCP/IP read, and that should be perfectly safe. (It is rare and unwelcome to have async code that cannot safely be interrupted).

Reply:
Yes the module really isn't necessary anymore. I removed it.